### ISSUE TITLE
BigDecimal / math operations with decimals with arbitrarily many digits

### ISSUE THREAD

**Issue by peheje** [dotnet/fsharp#14687]
**Describe the solution you'd like**

F# is marketed as a language good for Data Science and ML. Examples:

"F# for Data Science and ML F# is used in a variety of ways ... data science, and machine learning. ... are a great fit for data science and machine learning"
https://dotnet.microsoft.com/en-us/languages/fsharp

"F# is an excellent solution for programmatic data science"
https://fsharp.org/guides/data-science/

"Data analytics machine learning in F#"
https://learn.microsoft.com/en-us/events/dotnetconf-focus-on-fsharp/getting-started-with-data-analytics-machine-learning-in-fsharp

I would argue that F# or rather the dotnet platform is missing big feature to support this, a "BigDecimal" with arbitrarily many digits for mathematical operations. For example I stumpled upon [finding decimal-repeats in the reciprocal of primes](https://youtu.be/DmfxIhmGPP4), some of which happen after many thousands of decimals. Opening up my favorite language F# to get started, I was a bit disspointed. There's no easy way to do this on the dotnet platform that I know of, other than calling other libraries such as GMP or the like. 

Here's some code examples in Kotlin/JVM and Python how easy it is to work with, and then my attempt to workaround using System.Numerics.BigInteger in the end.

```kotlin
import java.math.BigDecimal
import java.math.RoundingMode
import kotlin.system.measureTimeMillis

fun main() {
    var result: BigDecimal?
    val executionTime = measureTimeMillis {
        val prime = BigDecimal(60013)
        result = BigDecimal.ONE.divide(prime, 1000000, RoundingMode.DOWN)
    }
    println(result)
    println("Execution time: ${executionTime}ms")
}
```
Takes about 200 ms on my PC. Python seems almost instant, I'm guessing it calls GMP:

```python
from decimal import *

getcontext().prec = 1000000
one = Decimal(1)
prime = Decimal(60013)
print(one / prime)
```

Doing it with bigint and multiply first works, takes around 1.5 seconds on my PC, but it's not very flexible to work with, if you really want a decimal number.

```fsharp
open System.Diagnostics

let prime = bigint 60013
let sw = Stopwatch.StartNew()
let zeros length = System.String('0', length)
let multiplier = bigint.Parse("1" + (zeros 1_000_000))
let ans = multiplier / prime
sw.Stop()
printfn "Execution time: %ims" sw.ElapsedMilliseconds
```

**In conclusion.**

Shouldn't F#/dotnet have a BigDecimal type?
Especially in the light of the data-science marketing.

**Comment by vzarytovskii** [comment]
This should go via the fslang-suggestions, and be reviewed there, I think.
Would you mind creating it there?

**Comment by vzarytovskii** [comment]
I am going to close this one as it is a much broader issue/request, which fits the best in the BCL, @peheje would you mind creating it in the runtime repo instead (https://github.com/dotnet/runtime/issues/new?assignees=&labels=api-suggestion&template=02_api_proposal.yml&title=%5BAPI+Proposal%5D%3A+)

**Comment by peheje** [comment]
I can create it there. But I do think the F# tag is warranted here in some sense due to the focus on data science? Wouldn't you agree?

**Comment by vzarytovskii** [comment]
> I can create it there. But I do think the F# tag is warranted here in some sense due to the focus on data science? Wouldn't you agree?

I wouldn't personally say so, it sure may be more convenient to use from F#, but C# and VB can benefit from it too.
